tests/test_train_taskcluster.py

```diff
@@ -76,8 +80,262 @@
+            2,
+            None,
+            ["scheduled", "rerun", "exception"],
+            {1: PARTIAL_ARTIFACTS, 0: PARTIAL_ARTIFACTS},
+            "",
+            "",
+            id="run_2_cant_continue_earlier_runs",
+        ),
+        pytest.param(
+            2,
+            None,
+            ["scheduled", "retry", "rerun"],
+            {1: PARTIAL_ARTIFACTS, 0: PARTIAL_ARTIFACTS},
+            "use",
+            "use",
+            id="run_2_cant_continue_earlier_runs_preserves_pretrained_model_mode",
+        ),
+    ),
+)
+def test_autocontinue(
+    current_run_id,
+    resumable_run_id,
+    reasons_created,
+    run_artifacts,
+    orig_pretrained_model_mode,
+    expected_pretrained_model_mode,
+):
+    with mock.patch.multiple(
+        train_taskcluster, subprocess=mock.DEFAULT, requests=mock.DEFAULT
+    ) as tt_mock:
+        with mock.patch.dict(os.environ) as mocked_env:
+            # In production, these are set by the Taskcluster worker
+            mocked_env["TASK_ID"] = "abcdef"
+            mocked_env["RUN_ID"] = str(current_run_id)
+            mocked_env["TASKCLUSTER_ROOT_URL"] = "https://some.cluster"
+
+            def fake_get(url, *args, **kwargs):
+                """Handles the expected requests to the Taskcluster API, and throws errors
+                for any requests that shouldn't have been made."""
+
+                resp = requests.Response()
+                if url.endswith("artifacts"):
+                    resp.status_code = 200
+                    resp.headers = {"Content-Type": "application/json"}
+                    run_id = int(url.split("/runs/", 1)[1].split("/")[0])
+                    resp._content = json.dumps(
+                        {
+                            "artifacts": run_artifacts[run_id],
+                        }
+                    ).encode("utf-8")
+                elif any(
+                    [
+                        url.endswith(artifact)
+                        for artifact in train_taskcluster.CONTINUATION_ARTIFACTS
+                    ]
+                ):
+                    # No action needed here; we will check that the right calls were
+                    # made based on the current_run_id later.
+                    resp.status_code = 200
+                    resp._content = b""
+                    resp.raw = io.StringIO("")
+                elif url.endswith("live.log") or url.endswith("live_backing.log"):
+                    resp.status_code = 400
+                    resp._content = (
+                        f"train_taskcluster.py wrongly tried to download a task log: {url}"
+                    )
+                else:
+                    resp.status_code = 400
+                    resp._content = f"train_taskcluster.py made a call to an unexpected URL: {url}"
+
+                return resp
+
+            tt_mock["requests"].get = mock.Mock(wraps=fake_get)
+
+            with tempfile.TemporaryDirectory() as model_dir:
```

Comment:
Suggestion: We have the `DataDir` fixture that puts these directories in the same place. This makes it easier debug failing tests.

```
data_dir = DataDir("test_train_taskcluster")
data_dir.path
```

Reply:
That's handy! Let's use it here too.